config/routes.rb

```diff
@@ -13,4 +13,21 @@
       resources :likes, only: [:create, :destroy]
     end
   end
+
+  # namespace :api do
+  #   namespace :v1 do
+  #     devise_for :users, :skip => [:registrations, :passwords, :confirmations ]
+  #     resources :posts, only: [:index]
+  #   resources :comments, only: [:index, :create]
+  namespace :api do
+    namespace :v1 do
+      resources :users do
+        resources :posts do
+          resources :comments
```

Comment:
**Optional**
- As the routes set up correctly, it can be better by specifying only the needed routes for each table. For example, you only have the `index` method so the `GET` API endpoint for the posts controller. You could do something like `resources :posts, only: [:index]` or
`get 'api/v1/user_id/posts', to: 'posts#index'`